go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/parca-dev/parca-agent
 
-go 1.17
+go 1.18
```

Comment:
This is actually not needed. This specified minimum required version to compile this module/codebase, if we don't need a specific version from 1.18 we shouldn't update this.

Reply:
I think we need it for https://github.com/parca-dev/parca-agent/pull/341 (https://tip.golang.org/doc/go1.18 says that `Settings` was made public in that release)

Reply:
Ah, then yes. Sorry I've missed that. It'd have been better to add these changes to that PR then.